pyproject.toml

```diff
@@ -1,3 +1,76 @@
+[build-system]
+requires = ["setuptools>=61.2"]
+build-backend = "setuptools.build_meta"
+
+[project]
+name = "trio"
+description = "A friendly Python library for async concurrency and I/O"
+authors = [{name = "Nathaniel J. Smith", email = "njs@pobox.com"}]
+license = {text = "MIT OR Apache-2.0"}
+keywords = [
+    "async",
+    "io",
+    "networking",
+    "trio",
+]
+classifiers = [
+    "Development Status :: 3 - Alpha",
+    "Framework :: Trio",
+    "Intended Audience :: Developers",
+    "License :: OSI Approved :: MIT License",
+    "License :: OSI Approved :: Apache Software License",
+    "Operating System :: POSIX :: Linux",
+    "Operating System :: MacOS :: MacOS X",
+    "Operating System :: POSIX :: BSD",
+    "Operating System :: Microsoft :: Windows",
+    "Programming Language :: Python :: Implementation :: CPython",
+    "Programming Language :: Python :: Implementation :: PyPy",
+    "Programming Language :: Python :: 3 :: Only",
+    "Programming Language :: Python :: 3.8",
+    "Programming Language :: Python :: 3.9",
+    "Programming Language :: Python :: 3.10",
+    "Programming Language :: Python :: 3.11",
+    "Programming Language :: Python :: 3.12",
+    "Topic :: System :: Networking",
+    "Typing :: Typed",
+]
+requires-python = ">=3.8"
+dependencies = [
+    # attrs 19.2.0 adds `eq` option to decorators
+    # attrs 20.1.0 adds @frozen
+    "attrs >= 20.1.0",
+    "sortedcontainers",
+    "idna",
+    "outcome",
+    "sniffio >= 1.3.0",
+    # cffi 1.12 adds from_buffer(require_writable=True) and ffi.release()
+    # cffi 1.14 fixes memory leak inside ffi.getwinerror()
+    # cffi is required on Windows, except on PyPy where it is built-in
+    "cffi>=1.14; os_name == 'nt' and implementation_name != 'pypy'",
+    "exceptiongroup >= 1.0.0rc9; python_version < '3.11'",
+]
+dynamic = ["version"]
+
+[project.readme]
+file = "LONG_DESCRIPTION.rst"
+content-type = "text/x-rst"
+
+[project.urls]
+Homepage = "https://github.com/python-trio/trio"
+Documentation = "https://trio.readthedocs.io/"
+Changelog = "https://trio.readthedocs.io/en/latest/history.html"
+
+[tool.setuptools]
+# This means, just install *everything* you see under trio/, even if it
+# doesn't look like a source file, so long as it appears in MANIFEST.in:
+include-package-data = true
+
+[tool.setuptools.packages]
+find = {namespaces = false}
+
+[tool.setuptools.dynamic]
+version = {attr = "trio._version.__version__"}
```

Comment:
this is the magic to avoid setuptools-scm

Reply:
Yea. Though, setuptools-scm provide many more things, than just generating a version.. So it's not exactly a replacement, just sourcing version from importables.